blog/views.py

```diff
@@ -67,10 +69,8 @@ def article_edit(id):
     if request.method == 'POST':
         article.title = request.form.get('title')
         article.content = request.form.get('content')
-        if bool(request.form.get('published')):
+        if bool(request.form.get('published')) and not article.publication_date:
             article.publication_date = datetime.datetime.utcnow()
```

Comment:
If you create an `article.published` property, you need to use it for publishing / unpublishing:

```python
article.published = bool(request.form.get('published'))
if article.published and not article.publication_date:
    article.publication_date = datetime.datetime.utcnow()
```

Reply:
I have to add in the model right ?

Reply:
You're totally right my python friend.